net/tests/Sails.DemoClient.Tests/DemoClientTests.cs

```diff
@@ -0,0 +1,214 @@
+using System;
+using System.Threading;
+using Microsoft.Extensions.DependencyInjection;
+using Sails.DemoClient.Tests._Infra.XUnit.Fixtures;
+using Sails.Remoting.Abstractions;
+using Sails.Remoting.Abstractions.Core;
+using Sails.Remoting.DependencyInjection;
+using Sails.Remoting.Options;
+using Substrate.Gear.Client.NetApi.Model.Types.Base;
+using Substrate.NetApi.Model.Types.Base;
+using Substrate.NetApi.Model.Types.Primitive;
+
+namespace Sails.DemoClient.Tests;
+
+public class DemoClientTests : IAssemblyFixture<SailsFixture>
+{
+    public DemoClientTests(SailsFixture sailsFixture)
+    {
+        this.sailsFixture = sailsFixture;
+        var serviceCollection = new ServiceCollection();
+        serviceCollection.AddRemotingViaNodeClient(
+            new NodeClientOptions
+            {
+                GearNodeUri = this.sailsFixture.GearNodeWsUrl,
+            });
+        var serviceProvider = serviceCollection.BuildServiceProvider();
+        this.remotingProvider = serviceProvider.GetRequiredService<IRemotingProvider>();
+        this.remoting = this.remotingProvider.CreateRemoting(SailsFixture.AliceAccount);
+
+    }
+    private static readonly Random Random = new((int)DateTime.UtcNow.Ticks);
+
+    private readonly SailsFixture sailsFixture;
+    private readonly IRemotingProvider remotingProvider;
+    private readonly IRemoting remoting;
+
+
+    [Fact]
+    public async Task Demo_DefaultConstructor_Works()
+    {
+        // arrange
+        var codeId = await this.sailsFixture.GetDemoContractCodeIdAsync();
+
+        // act
+        var demoFactory = new Demo.DemoFactory(this.remoting);
+        var activate = await demoFactory
+            .Default()
+            .ActivateAsync(codeId, BitConverter.GetBytes(Random.NextInt64()), CancellationToken.None);
+        var programId = await activate.ReceiveAsync(CancellationToken.None);
+
+        // assert
+        Assert.NotNull(programId);
+    }
+
+    [Fact]
+    public async Task Demo_Activation_Throws_NotEnoughGas()
+    {
+        // arrange
+        var codeId = await this.sailsFixture.GetDemoContractCodeIdAsync();
+
+        // act
+        var demoFactory = new Demo.DemoFactory(this.remoting);
+        var activate = await demoFactory
+            .Default()
+            .WithGasLimit(new GasUnit(0))
+            .ActivateAsync(codeId, BitConverter.GetBytes(Random.NextInt64()), CancellationToken.None);
+        // throws on ReceiveAsync
+        var ex = await Assert.ThrowsAsync<ArgumentException>(() => activate.ReceiveAsync(CancellationToken.None));
+
+        // assert
+        // TODO assert custom exception
+        Assert.NotNull(ex);
+    }
```

Comment:
It would nice to see these tests a bit split. These ones, for example, should be in the already created `DemoFactoryTests` class. For the ones, you can create test classes per service `CounterTests`, `PingPongTests`, etc. They can have a base class for the common arrange